openapi_tester/clients.py

```diff
@@ -22,14 +22,21 @@ def __init__(
     ) -> None:
         """Initialize ``OpenAPIClient`` instance."""
         super().__init__(*args, **kwargs)
+        #
         self.schema_tester = schema_tester or self._schema_tester_factory()
 
     def request(self, **kwargs) -> Response:  # type: ignore[override]
         """Validate fetched response against given OpenAPI schema."""
         response = super().request(**kwargs)
+        if self._is_successfull_response(response):
+            self.schema_tester.validate_request(response)
         self.schema_tester.validate_response(response)
         return response
 
+    @staticmethod
+    def _is_successfull_response(response: Response) -> bool:
+        return response.status_code in range(200, 300) and response.status_code != 204
```

Comment:
Is there a reason we cannot use `response.ok` or similar built-in functions in the requests response class?

Reply:
The `response.ok` returns `True` for anything < 400, but seems appropriate here? https://github.com/psf/requests/blob/main/src/requests/models.py#L755

Reply:
I tried this, and this is true for `requests` library `Response` type, but in this case, with `OpenAPIClient` the response we will get is of type `rest_framework.response.Response`. Unfortunately it doesn't count with a built in `ok` property, `raise_for_status` or similar 😢 .  I have modified the check though to consider everything `< 400` as successful.